client/my-sites/themes/current-theme/index.jsx

```diff
@@ -126,5 +147,6 @@ export default connect( ( state, { siteId } ) => {
 	return {
 		currentThemeId,
 		currentTheme: getCanonicalTheme( state, siteId, currentThemeId ),
+		isSiteUsingCoreSiteEditor: isSiteUsingCoreSiteEditorSelector( state, siteId ),
```

Comment:
We could straight up return `showBetaBadge` here instead of computing it in the rendered.
E.g.

```js
const currentTheme = getCanonicalTheme( state, siteId, currentThemeId );
return {
  currentTheme,
  showBetaBadge: isFullSiteEditingTheme( currentTheme ) && isSiteUsingCoreSiteEditor( state, siteId )
}
```

It would also improve the import naming (`isSiteUsingCoreSiteEditorSelector` makes it slightly harder to search), given we wouldn't destructure the prop anymore.